haunted-houses/src/main/java/cz/muni/fi/pa165/hauntedhouses/dao/impl/SpecterDaoImpl.java

```diff
@@ -1,6 +1,78 @@
 package cz.muni.fi.pa165.hauntedhouses.dao.impl;
 
 import cz.muni.fi.pa165.hauntedhouses.dao.SpecterDao;
+import cz.muni.fi.pa165.hauntedhouses.model.Ability;
+import cz.muni.fi.pa165.hauntedhouses.model.House;
+import cz.muni.fi.pa165.hauntedhouses.model.Specter;
 
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceContext;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * @author Zoltan Fridrich
+ */
+@Repository
+@Transactional
 public class SpecterDaoImpl implements SpecterDao {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public void createSpecter(Specter specter) {
+        em.persist(specter);
+    }
+
+    @Override
+    public void updateSpecter(Specter specter) {
+        em.merge(specter);
+    }
+
+    @Override
+    public void deleteSpecter(Specter specter) {
+        em.remove(specter);
```

Comment:
Toto máme tiež inak napísané. Remove nefunguje, ak je daná entita detached. Treba ju najskôr primergovať. Viď naše implementácie

Reply:
to som popravde nevedel ze naco to sluzi.